src/Scaffolding/VS.Web.CG.Mvc/BlazorIdentity/BlazorIdentityGenerator.cs

```diff
@@ -0,0 +1,544 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.DotNet.Scaffolding.Shared;
+using Microsoft.DotNet.Scaffolding.Shared.CodeModifier;
+using Microsoft.DotNet.Scaffolding.Shared.CodeModifier.CodeChange;
+using Microsoft.DotNet.Scaffolding.Shared.Project;
+using Microsoft.DotNet.Scaffolding.Shared.ProjectModel;
+using Microsoft.DotNet.Scaffolding.Shared.T4Templating;
+using Microsoft.VisualStudio.Web.CodeGeneration;
+using Microsoft.VisualStudio.Web.CodeGeneration.CommandLine;
+using Microsoft.VisualStudio.Web.CodeGeneration.DotNet;
+using Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore;
+using Microsoft.VisualStudio.Web.CodeGeneration.Utils;
+using Microsoft.VisualStudio.Web.CodeGenerators.Mvc.BlazorIdentity;
+using Microsoft.VisualStudio.Web.CodeGenerators.Mvc.Identity;
+using Microsoft.VisualStudio.Web.CodeGenerators.Mvc.Templates.General;
+using ConsoleLogger = Microsoft.DotNet.MSIdentity.Shared.ConsoleLogger;
+
+namespace Microsoft.VisualStudio.Web.CodeGenerators.Mvc.Blazor
+{
+    [Alias("blazor-identity")]
+    public class BlazorIdentityGenerator : ICodeGenerator
+    {
+        private IApplicationInfo AppInfo { get; set; }
+        private ILogger Logger { get; set; }
+        private IModelTypesLocator ModelTypesLocator { get; set; }
+        private IFileSystem FileSystem { get; set; }
+        private IFilesLocator FileLocator { get; set; }
+        private IProjectContext ProjectContext { get; set; }
+        private Workspace Workspace { get; set; }
+        private ConsoleLogger ConsoleLogger { get; set; }
+        private ICodeGenAssemblyLoadContext AssemblyLoadContextLoader { get; set; }
+        private const string Main = nameof(Main);
+        private bool CalledFromCommandline => !(FileSystem is SimulationModeFileSystem);
+
+        private IEnumerable<string> _templateFolders;
+        //Folders where the .tt templates for Blazor Identity scenario live. Should be in VS.Web.CG.Mvc\Templates\BlazorIdentity
+        private IEnumerable<string> TemplateFolders
+        {
+            get
+            {
+                if (_templateFolders is null || !_templateFolders.Any())
```

Comment:
For this condition, is the intention that if the `TemplateFoldersUtilities.GetTemplateFolders` call below returned an empty result set, that we'd keep checking it? Should that scenario even really happen? Or should this effectively just be a lazy load (if null check only, no empty check)?

I guess the same question applies to all the below collections as well (`AllBlazorIdentityFiles`, `GeneralT4Files`, `BlazorIdentityTemplateTypes`).

Reply:
it should just be a lazy load, null check only. We don't need to keep to checking, tyty.